### PR TITLE
Issue #6207: Add xpath regression test for ParameterName

### PR DESCRIPTION
**Pull request by 0xbakry**
Issue: https://github.com/checkstyle/checkstyle/issues/6207

This PR adds XPath regression test for suppression filter on [ParameterName](https://checkstyle.org/config_naming.html#ParameterName) check.

### CODE REVIEW

File: src/it/resources/org/checkstyle/suppressionxpathfilter/parametername/SuppressionXpathRegressionParameterName3.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.checkstyle.suppressionxpathfilter.parametername;
+
+class SuppressionXpathRegressionParameterName3{
+  void method2(int V2) {} // warn
+  @Override
+  public boolean equals(Object V3) { // OK
+    return true;
```
Comment (Vyom-Yadav) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/parametername/SuppressionXpathRegressionParameterName3.java:7]: Lowercase please, `// OK` -> `// ok`

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    private final String checkName = ParameterNameCheck.class.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void test1() throws Exception {
```
Comment (Vyom-Yadav) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java:42]: I know at many places we use `test1`, `test2`.....
Can you please name these test cases based on the properties of the check they are using, see https://github.com/checkstyle/checkstyle/blob/master/src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionAbstractClassNameTest.java for example.

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionParameterName1.java"));
```
Comment (Vyom-Yadav) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java:44]: Same for input files, can the name be adjusted according to the check's property being used?

**Review by Vyom-Yadav — CHANGES_REQUESTED** [review]
Items:

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java (hunk 2 above)
Comment (0xbakry) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java:42]: Done

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java (hunk 3 above)
Comment (0xbakry) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java:44]: Done.

**Comment by romani** [comment]
@0xbakry , from what geo region you are ? We noticed this circleci behavior like this if CI thinks you are from sanctioned regions. If you are, we will workaround it, if not we need to find reason why CI do not like your account

**Comment by 0xbakry** [comment]
@romani I'm from Egypt, i don't know if it's a sanctioned region or not

**Comment by romani** [comment]
It should not be. We had people from Egypt before.

**Comment by 0xbakry** [comment]
well, what should i do now?

**Comment by nrmancuso** [comment]
It is strange, sanctioned country workflow run looks like this: https://app.circleci.com/pipelines/github/checkstyle/checkstyle/17821/workflows/ef8b5ac6-4e5b-44fb-8b01-6277c60e4f88/jobs/263603

This can be found by filtering workflows by "Not Run". However, I do not see any in this status for @0xbakry.

**Comment by 0xbakry** [comment]
Did this happen in https://github.com/checkstyle/checkstyle/pull/12916/ too?

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java (hunk 2 above)
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java]: Resolved.

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java (hunk 3 above)
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionParameterNameTest.java]: Resolved.

File: src/it/resources/org/checkstyle/suppressionxpathfilter/parametername/SuppressionXpathRegressionParameterNameIgnoreOverridden.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package org.checkstyle.suppressionxpathfilter.parametername;
+
+class SuppressionXpathRegressionParameterNameIgnoreOverridden{
+  void method2(int V2) {} // warn
+  @Override
+  public boolean equals(Object V3) { // ok
+    return true;
+  }
+}
```
Comment (Vyom-Yadav) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/parametername/SuppressionXpathRegressionParameterNameIgnoreOverridden.java:9]: Please add some line breaks in between, no need to compress so much, you can regularly format your code. Please do it for other files too.

Comment (nrmancuso) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/parametername/SuppressionXpathRegressionParameterNameIgnoreOverridden.java:9]: @0xbakry please separate methods and class definition by empty line. The only reason that we do not have a check on this for test input files is that sometimes we require poorly formatted input for test cases. If we do not have such a requirement for poorly formatted code for this test input, format code to make it easy to read.

Comment (0xbakry) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/parametername/SuppressionXpathRegressionParameterNameIgnoreOverridden.java]: I got your point, thanks.
I updated it.

Comment (Vyom-Yadav) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/parametername/SuppressionXpathRegressionParameterNameIgnoreOverridden.java]: Resolved.

**Review by Vyom-Yadav — APPROVED** [review]
LGTM! Thanks a lot, @0xbakry

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**